### ISSUE TITLE
Rule S3649 needs ability to prove sanitization

### ISSUE THREAD

**Issue by SCordesHum** [SonarSource/sonar-csharp#771]
### Description

If we hardcode the stored procedure name in CommandText, the warning is correctly not given.  If, however, we read the stored procedure name from configuration and then assign it to the CommandText property, the warning is given.  That is understandable, under the rubric of "Don't trust your own configuration values."  

But there does not seem to be a way to sanitize the 'variable' value that satisfies the Rule.  Even though the CommandText value is variable as a whole, it has no variable parts in it,  Therefore, it is not subject to injection anymore than a hardcoded value would have been.

### Repro steps
```C#
string storedProcedureName = ... read configuration value ...;
SqlCommand sqlCommand = new SqlCommand(storedProcedureName, sqlConnection);
sqlCommand.CommandType = CommandType.StoredProcedure;

SqlParameter parameterOne = new SqlParameter("@ParamOne", "My param value");
sqlCommand.Parameters.Add(parameterOne);
 
sqlCommand.ExecuteNonQuery();
```

### Expected behavior

There needs to be a way to sanitize the value, i.e., to prove to the Rule that the value was sanitized.

### Actual behavior

The warning is given, even if CommandType = CommandType.StoredProcedure.

### Known workarounds

None.  Hardcoding is not desirable, as it limits maintainability.

### Related information

* SonarC# 6.2.0.2536
* Visual Studio 2015
